context/app/static/js/components/savedLists/SavedListPanel/hooks.ts

```diff
@@ -1,18 +1,43 @@
-import useSavedEntityData from 'js/hooks/useSavedEntityData';
+import { useMemo } from 'react';
 import { SavedEntity } from 'js/components/savedLists/types';
-import { getSearchHitsEntityCounts } from 'js/helpers/functions';
+import { getIDsQuery } from 'js/helpers/queries';
+import useSearchData from 'js/hooks/useSearchData';
 
-const source = ['entity_type']; // 'memoize' to prevent infinite loop
+interface EntityCounts {
+  entity_counts: {
+    buckets: { doc_count: number; key: string }[];
+  };
+}
+function useSavedEntityCountsData(savedEntities: Record<string, SavedEntity>) {
+  const query = useMemo(
+    () => ({
+      size: 0,
+      query: getIDsQuery(Object.keys(savedEntities)),
+      aggs: {
+        entity_counts: {
+          terms: { field: 'entity_type.keyword', size: 10000 },
+        },
+      },
+    }),
+    [savedEntities],
+  );
+
+  const { searchData, isLoading } = useSearchData<unknown, EntityCounts>(query);
+  return { searchData, isLoading };
+}
 
 function useSavedEntityTypeCounts(listSavedEntities: Record<string, SavedEntity>) {
-  const { searchHits } = useSavedEntityData(listSavedEntities, source);
-  const counts = getSearchHitsEntityCounts(searchHits);
+  const { searchData } = useSavedEntityCountsData(listSavedEntities);
 
-  return {
-    donors: counts.Donor || 0,
-    samples: counts.Sample || 0,
-    datasets: counts.Dataset || 0,
-  };
+  const counts = { donors: 0, samples: 0, datasets: 0 };
+
+  searchData?.aggregations?.entity_counts?.buckets?.forEach((bucket) => {
+    if (bucket.key === 'Donor') counts.donors = bucket.doc_count;
+    else if (bucket.key === 'Sample') counts.samples = bucket.doc_count;
+    else if (bucket.key === 'Dataset') counts.datasets = bucket.doc_count;
+  });
```

Comment:
This should work! I'm a bit of a stickler for minimizing mutations outside of hooks, so I took a stab at wrapping this in a useMemo (might need some linting/formatting fixes):

```suggestion
  const counts = useMemo(() => {
  	return searchData?.aggregations?.entity_counts?.buckets?.reduce((acc, bucket) => { 
	  	const { key, doc_count } = bucket;
	  	switch(key) {
	  		case "Donor": 
	  			acc.donors = doc_count;
	  			break;
				case "Sample":
					acc.samples = doc_count;
					break;
				case "Dataset":
					acc.datasets = doc_count;
					break;
	  	}
	    return acc;
	  }, 
	  { donors: 0, samples: 0, datasets: 0 }, [searchData])
```